fvm/errors/errors.go

```diff
@@ -102,54 +138,167 @@ func SplitErrorTypes(inp error) (err CodedError, failure CodedError) {
 
 // HandleRuntimeError handles runtime errors and separates
 // errors generated by runtime from fvm errors (e.g. environment errors)
-func HandleRuntimeError(err error) error {
-	if err == nil {
+func HandleRuntimeError(inp error) (err error) {
+	if inp == nil {
 		return nil
 	}
 
 	// if is not a runtime error return as vm error
 	// this should never happen unless a bug in the code
-	runErr, ok := err.(runtime.Error)
+	runErr, ok := inp.(runtime.Error)
 	if !ok {
 		return NewUnknownFailure(err)
 	}
 
```

Comment:
once https://github.com/onflow/cadence/pull/3052 is merged, this method could be simplified back to returning `NewCadenceRuntimeError(runErr)` here